webpack/assets/javascripts/react_app/components/common/TypeAheadSelect/TypeAheadSelectReducer.js

```diff
@@ -0,0 +1,24 @@
+import Immutable from 'seamless-immutable';
+import { UPDATE_OPTIONS, UPDATE_SELECTED } from './TypeAheadSelectConstants';
+
+const initialState = Immutable({});
+
+export default (
+  state = initialState,
+  { type, payload: { id, options, items } = {} }
+) => {
+  switch (type) {
+    case UPDATE_OPTIONS:
+      return state.setIn([id], {
+        ...state[id],
+        options,
+      });
+    case UPDATE_SELECTED:
+      return state.setIn([id], {
+        ...state[id],
+        selected: items.map(item => (item.label ? item.label : item)),
```

Comment:
Can we move `selected` into a selector?
And maybe change the Action type to UPDATE_ITEMS

Reply:
Do you mean calling a selector from here? Also, the prop is called `selected` so maybe it's better to rename `items` to `selected` ?

Reply:
Gotcha, thanks a rename would be great !